server.py

```diff
@@ -243,6 +244,35 @@ def do_get_response(tx_id):
         return jsonify({}), 404
 
 
+@app.route('/responses/delete-old', methods=['POST'])
+def delete_old_responses():
+    """Deletes responses that are older than the number of days set in config
+    Config use is a compromise for safety in case incorrect parameters are passed.
+    """
+    response_retention_days = settings.RESPONSE_RETENTION_DAYS
+
+    if not response_retention_days:
+        return server_error('Response retention days not configured')
+
+    # Get the cut off date
+    time_delta = datetime.timedelta(days=response_retention_days)
+    cut_off_date = datetime.datetime.utcnow() - time_delta
+
+    # Set time element of cut off date to 0:0:0.0
+    cut_off_date = cut_off_date.combine(cut_off_date.date(),
+                                        datetime.time(hour=0, minute=0, second=0, microsecond=0, tzinfo=None))
+
+    try:
+        deleted_count = db.session.query(SurveyResponse).filter(SurveyResponse.ts < cut_off_date) \
+            .delete(synchronize_session=False)
```

Comment:
What's this flag for?

Reply:
synchronize_session –

chooses the strategy for the removal of matched objects from the session. Valid values are:

False - don’t synchronize the session. This option is the most efficient and is reliable once the session is expired, which typically occurs after a commit(), or explicitly using expire_all(). Before the expiration, objects may still remain in the session which were in fact deleted which can lead to confusing results if they are accessed via get() or already loaded collections.

'fetch' - performs a select query before the delete to find objects that are matched by the delete query and need to be removed from the session. Matched objects are removed from the session.

'evaluate' - Evaluate the query’s criteria in Python straight on the objects in the session. If evaluation of the criteria isn’t implemented, an error is raised.

Given we are doing a bulk delete and not using an ORM , this is the better option